src/AzureClient/Magic/ExecuteMagic.cs

```diff
@@ -52,6 +52,8 @@ public ExecuteMagic(IAzureClient azureClient, ILogger<ExecuteMagic> logger)
 
                         - `{AzureSubmissionContext.ParameterNameJobName}=<string>`: Friendly name to identify this job. If not specified,
                         the Q# operation or function name will be used as the job name.
+                        - `{AzureSubmissionContext.ParameterNameJobParams}=< ImmutableDictionary<string, string>>`: Provider-specific job parameters
```

Comment:
The type `ImmutableDictionary<string, string>` has no meaning in this context; readers will be working either in a Q# notebook or in a Q# + Python workflow, neither of which has an exact mapping for that C# type. In general, the types listed in this documentation string are really intended to be informal summaries for users as to what kind of input users can expect to pass.

Reply:
Looking at unit tests now -- thought I'd get this initial code out there for comments while I worked on them.  I will address the Q# + Python workflow requirements next.

Regarding the references to ImmutableDictionary<string, string> in the documentation:  I was following the pattern used for the other parameters.  Should a simplified description of ImmutableDictionary<string, string> be substituted, or should that data type not be mentioned at all?

Reply:
> Looking at unit tests now -- thought I'd get this initial code out there for comments while I worked on them.

Makes sense, but in that case I'd suggest converting this to a draft PR so that it doesn't accidentally get merged before unit tests are added and the Q# + Python interop interface is modified for this new feature.

> Regarding the references to ImmutableDictionary<string, string> in the documentation: I was following the pattern used for the other parameters. Should a simplified description of ImmutableDictionary<string, string> be substituted, or should that data type not be mentioned at all?

The other entries use informal descriptions like "string" and "integer," since there's not any formal notion of types when calling magic commands from within a Q# notebook. In that sense, magic commands work a lot like command-line options (`--foo=bar`), such that simplified descriptions work well together with the more explicit examples later in the command documentation.

Reply:
- Feature is supported in the Q# + Python interop interface
- Unit tests have been added (C# and Python)
- Documentation has been simplified and clarified